### PR TITLE
Introduce `--format=ids` in user application password list

### PR DESCRIPTION
**Pull request by ernilambar**
Fixes https://github.com/wp-cli/entity-command/issues/474

### CODE REVIEW

File: src/User_Application_Password_Command.php [hunk 1]
```diff
@@ -197,8 +198,13 @@ static function ( $application_password ) use ( $field, $value ) {
...
 
-		$formatter = new Formatter( $assoc_args, $fields );
-		$formatter->display_items( $application_passwords );
+		if ( ! empty( $assoc_args['format'] ) && 'ids' === $assoc_args['format'] ) {
+			$formatter = new Formatter( $assoc_args, $fields );
+			$formatter->display_items( wp_list_pluck( $application_passwords, 'uuid' ) );
+		} else {
+			$formatter = new Formatter( $assoc_args, $fields );
+			$formatter->display_items( $application_passwords );
+		}
```
Comment (swissspidy) [inline src/User_Application_Password_Command.php:207]: Nit: move `new Formatter()` up before the condition:

```suggestion
		$formatter = new Formatter( $assoc_args, $fields );

		if ( ! empty( $assoc_args['format'] ) && 'ids' === $assoc_args['format'] ) {
			$formatter->display_items( wp_list_pluck( $application_passwords, 'uuid' ) );
		} else {
			$formatter->display_items( $application_passwords );
		}
```

Aside: I noticed many commands just do `echo implode( ' ', wp_list_pluck( ... ) );` instead 🤔  Not sure why. Using the formatter is cleaner.

Comment (ernilambar) [reply src/User_Application_Password_Command.php]: I agree. Those codes are from 7 or 8 years ago. Earlier such approach was used to print ids.

Comment (ernilambar) [reply src/User_Application_Password_Command.php:207]: The change you mentioned did not work. So I have kept as it is.

Comment (swissspidy) [reply src/User_Application_Password_Command.php:207]: How does moving `$formatter = new Formatter( $assoc_args, $fields );` up not work? 🤔 
It just removes a duplicated line by moving it out of the conditional block.

Comment (ernilambar) [reply src/User_Application_Password_Command.php:207]: In Formatter value is passed by reference. So it messes up the actual value.

**Comment by ernilambar** [comment]
@swissspidy In Formatter, `$assoc_args` is passed by reference. 

```
public function __construct( &$assoc_args, $fields = null, $prefix = false ) {
```
And it unsets the `format` key in the method. https://github.com/wp-cli/wp-cli/blob/main/php/WP_CLI/Formatter.php#L46

So, if we keep `$formatter = new Formatter( $assoc_args, $fields );` before the conditional, `if` condition will never be true because we now dont have `$assoc_args['format']`.

**Comment by ernilambar** [comment]
cc @danielbachhuber

**Comment by schlessera** [comment]
@ernilambar I suggest to go with the tweak that @swissspidy mentioned but also combine it with an early processing of the flag like so:
```
$format = Utils\get_flag_value( $assoc_args, 'format', 'table );
```
This will avoid the breakage with @swissspidy 's refactor, and it will also make the conditional(s) cleaner.

**Review by swissspidy — APPROVED** [review]

**Pull request merged**